savoirfairelinux/sous-chef
# Fix #432 (url translation)

## Fixes #432  by lingxiaoyang

### Changes proposed in this pull request:

* Add translatable strings in all `urls.py`.
* Add language prefix `/fr/` in URL for French. 
* Add translatable URL name for client creation form wizard.
* Add new test setting file, for running the unit test in French.
* Fix problems in French unit test.
* Add French unit test to Travis CI configuration.

### Status

- [ ] HOLD: need rebase after merging PR #601 

### How to verify this change

1. Visit `http://localhost:8000/fr/`, it should be French.
2. Visit every page, the URLs should be in French now (not all, because they need translation in `.po` files)
3. Go to client creation form wizard: the URL should show the steps in French.
4. Execute French unit test by: 
````
python3 src/manage.py test --settings=sous_chef.settings_test_fr
````
5. Go to Travis CI: the build log should include the French unit test.

Verify English:
1. Visit `http://localhost:8000`, it should be in English by default.
2. Now the URLs should be all in English

### Deployment notes and migration

none

### New translatable strings


### Additional notes

none
